lib/SILGen/SILGen.cpp

```diff
@@ -689,9 +689,11 @@ static bool shouldEmitFunctionBody(const AbstractFunctionDecl *AFD) {
     return false;
 
   auto &ctx = AFD->getASTContext();
-  if (ctx.TypeCheckerOpts.EnableLazyTypecheck) {
+  if (ctx.TypeCheckerOpts.EnableLazyTypecheck || AFD->isInMacroExpansionFromClangHeader()) {
     // Force the function body to be type-checked and then skip it if there
-    // have been any errors.
+    // have been any errors. Normally macro expansions are type checked in the module they
+    // expand in - this does not apply to swift macros applied to nodes imported from clang,
+    // so force type checking of them here if they haven't already, to prevent crashing.
```

Comment:
does this mean something further downstream is calling getBody() instead of getTypecheckedBody()? We should be able to get rid of this.

Reply:
I think the issue is that SILGen can't tell whether *this* function had an error in it, and so it doesn't avoid walking the erroneous AST. If we had that, we could effectively SILGen the body to a fatalError("type error") and keep going safely.